app/models/atmosphere/action.rb

```diff
@@ -0,0 +1,22 @@
+module Atmosphere
+  class Action < ActiveRecord::Base
+    belongs_to :appliance,
+               class_name: 'Atmosphere::Appliance'
+
+    has_many :action_logs, -> { order(id: :asc) },
+             dependent: :destroy,
```

Comment:
I'm not actually sure I understand the logic here. There's a `dependent: :destroy` on Appliance -> Action and likewise on Action -> ActionLog so once the appliance is gone, so are all its logs. Are you *absolutely sure* that's what you want to happen? It seems really weird; I would have expected a `dependent: :nullify` here...
Regarding your other comment - I don't think automatically destroying any appliances (regardless of their age) is a good idea. In Share people have been running appliances for months and still end up needing them.